File: cases/taint/java/argument-position-positive/ArgumentPositionPositive.java
package dataflowbench.taint;

final class ArgumentPositionPositive {
    static int dfb_source() { // DFB-SOURCE: argument-position-input
        return 1;
    }

    static int chooseFirst(int first, int second) { // DFB-WITNESS: argument-position-first
        return first;
    }

    static void dfb_sink(int value) { } // DFB-SINK: argument-position-sink

    static void run() {
        int result = chooseFirst(dfb_source(), 0);
        dfb_sink(result);
    }
}

File: cases/taint/kotlin/argument-position-positive/ArgumentPositionPositive.kt
package dataflowbench

object ArgumentPositionPositive {
    fun dfb_source(): Int { // DFB-SOURCE: argument-position-input
        return 1
    }

    fun chooseFirst(first: Int, second: Int): Int { // DFB-WITNESS: argument-position-first
        return first
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: argument-position-sink

    fun run() {
        val result = chooseFirst(dfb_source(), 0)
        dfb_sink(result)
    }
}
